### PR TITLE
#163519109  share article social 

### PR DESCRIPTION
**Pull request by NkFab**
#### What does this PR do?
Implement sharing an article.

#### Description of Task to be completed?
- Share an article on Facebook
- Share an article on Twitter
- Share an article on Linkedin
- Share an article in Email

#### How should this be manually tested?
- Clone this repo
- Run `npm install` or `yarn` to install dependencies 
- Run `npm run start` or `yarn start` to start the application
- Read an article scroll down at the foot of the article, click on either of the share icons depending on which platform you choose to share.

#### Any background context you want to provide?
- Facebook doesn't allow to share localhost links

#### What are the relevant pivotal tracker stories?
#163519109

### CODE REVIEW

**Comment by NkFab** [comment]
I am doing the rebasing process. This is not the final PR.

File: src/__tests__/actions/articleActions.test.js [hunk 1]
```diff
@@ -0,0 +1,309 @@
+import configureStore from 'redux-mock-store'; // ES6 modules
```
Comment (mcaleb808) [inline src/__tests__/actions/articleActions.test.js]: @NkFab try to remove this single line comment!

Comment (NkFab) [reply src/__tests__/actions/articleActions.test.js]: Done

File: src/__tests__/components/Article.test.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    test('should call onSelectedRating method instance', () => {
+      wrapper.find('button[data-value="5"]').simulate('click');
+      expect(props.rateArticle).toHaveBeenCalled();
+    });
+  });
+
+  describe("when clicking article's rates", () => {
+    beforeEach(() => {
+      wrapper = mount(<Article {...props} />);
+    });
```
Comment (oesukam) [inline src/__tests__/components/Article.test.js]: @NkFab Can you add an empty line after this line for better readability?

File: src/__tests__/components/Article.test.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      props.article.rated = 5;
+      wrapper = mount(<Article {...props} />);
+      expect(wrapper.props().article.rated).toBe(5);
+    });
+  });
+
+  describe('when clicking on rateArticle', () => {
+    beforeEach(() => {
+      wrapper = mount(<Article {...props} />);
+    });
```
Comment (oesukam) [inline src/__tests__/components/Article.test.js]: @NkFab Can you add an empty line after this line for better readability?

**Review by mcaleb808 — APPROVED** [review]

File: src/__tests__/components/ArticleCard.test.js [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
+  loading: true,
+  article: articleData,
+  url: '/articles/article-slug',
+  history: { push: jest.fn() },
+};
 
 describe('<ArticleCard />', () => {
+  beforeEach(() => {
+    wrapper = mount(<ArticleCard {...props} />);
+  });
```
Comment (oesukam) [inline src/__tests__/components/ArticleCard.test.js]: @NkFab Can you add an empty line after this line for better readability?

File: src/__tests__/components/Ratings.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      articleSlug: 'article-slug',
+    },
+  },
+  history: { push: jest.fn() },
+};
+
+describe('<Ratings />', () => {
+  beforeEach(() => {
+    wrapper = mount(<Ratings {...props} />);
+  });
```
Comment (oesukam) [inline src/__tests__/components/Ratings.test.js]: @NkFab Can you add an empty line after this line for better readability?

File: src/__tests__/components/common/Toast.test.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+let wrapper;
+const props = {
+  show: true,
+};
+jest.useFakeTimers();
+
+describe('<Toast />', () => {
+  beforeEach(() => {
+    wrapper = mount(<Toast {...props} />);
+  });
```
Comment (oesukam) [inline src/__tests__/components/common/Toast.test.js]: @NkFab Can you add an empty line after this line?

File: src/__tests__/components/Article.test.js [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+      articleSlug: 'article-slug',
+    },
+  },
+  history: { push: jest.fn() },
+};
+
+describe('<Article />', () => {
+  beforeEach(() => {
+    wrapper = mount(<Article {...props} />);
+  });
```
Comment (oesukam) [inline src/__tests__/components/Article.test.js]: @NkFab Can you add an empty line after this line for better readability?

File: src/components/Rating/RatingCard.jsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import moment from 'moment';
+import defaultAvatar from '../../assets/images/avatar.png';
+
+const RatingCard = ({ rate }) => (
+  <div className="rating-card">
+    <div className="rating-card__left">
+      <div className="rating">
+        <button className={rate.rating === 5 ? 'selected' : ''} />
+        <button className={rate.rating === 4 ? 'selected' : ''} />
+        <button title="text" className={rate.rating === 3 ? 'selected' : ''} />
```
Comment (oesukam) [inline src/components/Rating/RatingCard.jsx]: @NkFab Are you using the title's attribute anywhere? If not, it should be removed as it is not part of the `button's tag` attributes.

**Review by oesukam — CHANGES_REQUESTED** [review]

File: src/components/Rating/RatingCard.jsx (hunk 8 above)
Comment (NkFab) [reply src/components/Rating/RatingCard.jsx]: Done

File: src/__tests__/components/Article.test.js (hunk 7 above)
Comment (NkFab) [reply src/__tests__/components/Article.test.js]: Done

File: src/__tests__/components/common/Toast.test.js (hunk 6 above)
Comment (NkFab) [reply src/__tests__/components/common/Toast.test.js]: Done

File: src/__tests__/components/Ratings.test.js (hunk 5 above)
Comment (NkFab) [reply src/__tests__/components/Ratings.test.js]: Done

File: src/__tests__/components/ArticleCard.test.js (hunk 4 above)
Comment (NkFab) [reply src/__tests__/components/ArticleCard.test.js]: Done

File: src/__tests__/components/Article.test.js (hunk 3 above)
Comment (NkFab) [reply src/__tests__/components/Article.test.js]: Done

File: src/__tests__/components/Article.test.js (hunk 2 above)
Comment (NkFab) [reply src/__tests__/components/Article.test.js]: Done

File: webpack.config.js [hunk 9]
```diff
@@ -20,7 +20,7 @@ module.exports = {
         include: path.join(__dirname, 'src', 'assets'),
         loader: 'url-loader',
         options: {
-          limit: 8192, // in bytes
+          limit: 8192,
```
Comment (oesukam) [inline webpack.config.js:23]: @NkFab Why did you remove that comment? I think that it is useful because only having that limit does not tell you if it is in Mb's, Kb's and so on.

**Review by oesukam — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 9 above)
Comment (NkFab) [reply webpack.config.js:23]: That makes sense.

**Review by oesukam — APPROVED** [review]

File: src/components/Article/Article.jsx [hunk 10]
```diff
@@ -98,6 +103,15 @@ export class Article extends Component {
     e.preventDefault();
   };
 
+  SocialShare = (on) => {
```
Comment (kimotho-njoki) [inline src/components/Article/Article.jsx:147]: you can remove the brackets

Comment (NkFab) [reply src/components/Article/Article.jsx:147]: I tried to add this rule `"arrow-parens": ["error", "as-needed"]` in the eslint config file but it was removing parentheses from dispatch functions as well so I chose to leave it like that. Is there any other workaround?
